src/libs/github/file_info.ts

```
@@ -0,0 +1,64 @@
+import { Observable, of, zip } from 'rxjs'
+import { map, switchMap } from 'rxjs/operators'
+import { resolveRev, retryWhenCloneInProgressError } from '../../shared/repo/backend'
+import { FileInfo } from '../code_intelligence'
+import { getDeltaFileName, getDiffResolvedRev, parseURL } from './util'
+
+export const resolveDiffFileInfo = (codeView: HTMLElement): Observable<FileInfo> =>
+    of(codeView).pipe(
+        map(codeView => {
+            const { repoPath } = parseURL()
+
+            return { codeView, repoPath }
+        }),
+        map(({ codeView, ...rest }) => {
+            const { headFilePath, baseFilePath } = getDeltaFileName(codeView)
+            if (!headFilePath) {
+                throw new Error('cannot determine file path')
+            }
+
+            return { ...rest, codeView, headFilePath, baseFilePath }
+        }),
+        map(({ codeView, ...rest }) => {
+            const diffResolvedRev = getDiffResolvedRev()
+            if (!diffResolvedRev) {
+                throw new Error('cannot determine delta info')
```

In what case does this happen? This error message is rather unhelpful, should `getDiffResolvedRev` maybe throw the error instead?

---

There isn't an error as far as I know. This piece was just ported over from existing logic. Would probably be worth looking into at some point.